src/deck.ts

```diff
@@ -30,65 +29,82 @@ export function splitText(outString: string, cap = 2000): string[] {
 	return outStrings;
 }
 
-async function convertCard(card: DataCard): Promise<Card> {
+async function convertCard(card: Card): Promise<ICard> {
 	const status = await card.status;
-	const scopeReg = /([a-zA-Z]+): (\d)/g;
-	const statusMap: { [scope: number]: number } = {};
-	let result = scopeReg.exec(status);
+	const scopeRegex = /([a-zA-Z]+): (\d)/g;
+	let limitTCG = NaN,
+		limitOCG = NaN;
+	let result = scopeRegex.exec(status);
 	while (result !== null) {
 		const scope = result[1];
-		const count = parseInt(result[2], 10); // 2 capture groups ensured by regex
-		// 3 copies is the default fallback
-		if (count < 3) {
-			// TODO: Less hardcode
-			if (scope === "OCG") {
-				statusMap[0x1] = count;
-			} else if (scope === "TCG") {
-				statusMap[0x2] = count;
-			}
+		const count = parseInt(result[2], 10);
+		if (scope === "OCG") {
+			limitOCG = count;
+		} else if (scope === "TCG") {
+			limitTCG = count;
 		}
-		result = scopeReg.exec(status);
+		result = scopeRegex.exec(status);
 	}
-	return new Card(card.text.en.name, card.data.ot, card.data.type, card.data.setcode, statusMap);
+	return {
+		name: card.text.en.name,
+		type: card.data.type,
+		setcode: card.data.setcode,
+		alias: card.data.alias || undefined,
+		limitTCG,
+		limitOCG,
+		isPrerelease: !!(card.data.ot & 0x100)
```

Comment:
We've got the ygoData enums included here so this could be `& enums.ot.OT_PRERELEASE`

Reply:
That is in fact not a thing! 😂

Reply:
Heck. Can't wait to get this whole thing cleaned up.